File: DocsRevision/DocsRevision/Entities/Document.swift
import Foundation

class Document{
    var Id: String
    var Name: String
    var FullName: String
    var CreationDate: NSDate
    var Extension: String
    var CreatorId: String
    var CreatorName: String
    var RevisionId: String
    var Revision: Revision
    
    init(
        id: String,
        name: String,
        fullName : String,
        creationDate: NSDate,
        exte: String,
        creatorId: String,
        creatorName: String,
        revisionId: String,
        revision: Revision) {
        self.Id=id
        self.Name=name
        self.FullName = fullName
        self.CreationDate=creationDate
        self.Extension = exte
        self.CreatorId=creatorId
        self.CreatorName=creatorName
        self.RevisionId=revisionId
        self.Revision=revision
    }
}

File: DocsRevision/DocsRevision.Xamarin/DocsRevision/Models/Document.cs
using System;

namespace DocsRevision.Models
{
    public class Document
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string NameFull { get; set; }
        public string Extension { get; set; }
        public DateTime CreationDate { get; set; }
        public string CreatorId { get; set; }
        public string CreatorName { get; set; }
        public string RevisionId { get; set; }
        public string RevisorName { get; set; }
        public Revision CurrentRevision { get; set; }
        public Revision LastRevision { get; set; }


        public string RevisionFormated { get => "Revisão " + CurrentRevision.NumRevision; }
        public string CreatedInFormated { get => "Criado em " + CreationDate.ToString("dd/MM/yyyy"); }
        public string CreatedForFormated { get => "Criado por " + CreatorName; }
        public string RevisorFormated { get => "Revisor " + RevisorName; }
        public string IsObsolet { get => CurrentRevision.IsObsolete ? "Obsoleto: Sim" : "Obsoleto: Não"; }
        public bool IsPendent { get => CurrentRevision.Status == Enum.EStatus.Pending ? false : true; }
        public bool IsNotPendent => !IsPendent;
        public bool IsRevisor { get => CurrentRevision.RevisorId == App.User.Id ? true : false; }
    }
}
